src/panels/config/areas/ha-config-area-page.ts

```diff
@@ -136,10 +144,60 @@ class HaConfigAreaPage extends LitElement {
       this.entities
     );
 
-    const grouped = groupBy(entities, (entity) =>
+    // Pre-compute the entity and device names, so we can sort by them
+    if (devices) {
+      devices.forEach((entry) => {
+        entry.name = computeEntityRegistryName(this.hass, entry);
+      });
+      sortDeviceRegistryByName(devices);
+    }
+    if (entities) {
+      entities.forEach((entry) => {
+        entry.name = computeEntityRegistryName(this.hass, entry);
+      });
+      sortEntityRegistryByName(entities);
+    }
+
+    // Group entities by domain
+    const groupedEntities = groupBy(entities, (entity) =>
       computeDomain(entity.entity_id)
     );
 
+    // Pre-compute the name also for the grouped and related entities so we can sort by them
+    let groupedAutomations: NameAndEntity<AutomationEntity>[] = [];
+    let groupedScenes: NameAndEntity<SceneEntity>[] = [];
+    let groupedScripts: NameAndEntity<ScriptEntity>[] = [];
+    let relatedAutomations: NameAndEntity<AutomationEntity>[] = [];
+    let relatedScenes: NameAndEntity<SceneEntity>[] = [];
+    let relatedScripts: NameAndEntity<ScriptEntity>[] = [];
+
+    if (isComponentLoaded(this.hass, "automation")) {
+      const data = this._prepareEntities<AutomationEntity>(
```

Comment:
I did not manage to get the unpack directly into the above defined vars working, so hence the extra step via the `data` var. I assume there is a more elegant way, but I did not find it 😞